Services/AssessmentQuestion/docs/readme.md

```diff
@@ -0,0 +1,99 @@
+# Introduction
+
+The implementation of assessment questions needs to be decoupled from the large component Test and Assessment. The codebase is to be extracted to its own service AssessmentQuestion.
+
+This concept describes the interfaces that will be introduced and how they are to be used by developers who want to integrate assessment questions to their components. Furthermore it describes how the database gets decoupled since the supposed separation in two different table spaces within the current state does not reflect a neccessary strict distinction.
+
+The AssessmentQuestion service is designed as a component that offers complex functionality but it keeps stupid for itself. The way other components can integrate assessment questions keeps as most flixible as possible. This strategy makes it possible to define most of the assessment logic within a consumer.
+
+This concept keeps focus on the decoupling, not on fullfilling future requirements.
+
+# Service Interfaces
+
+The AssessmentQuestion service will come with the following interfaces that can be used by other developers that want to integrate assessment questions to their component. The current obect structure consisting of a GUI class and an Object class gets overhauled.
+
+Objects implementing `ilAsqQuestion` represents what was also formaly known as the Object class while objects implementing `ilAsqQuestionAuthoring` are about the `executeCommand` structure that was formly knwon as the GUI classes. All other aspects have been extracted from these interfaces.
+
+The interface `ilAsqPresentation` provides all functionality to output a question and its additional contents. These parts are fully extracted from the former GUI class. Solutions now get injected to keep the presentation as modular as possible.
+
+The interface `ilAsqResultCalculator` provides all functionality of calculating right/wrong for a given solution as well as reached points. Having this functionality in an own object following implementing this interface makes it possible for consumers to surround this kind calculators with an own proxy calculator implementing the same interface (e.g.for any score cutting options).
+
+The handling of solutions is defined by the interface `ilAsqSolution` so the future implementation is fully getting rid of dealing with solution values stored in row array structures that were queried from the database.
```

Comment:
Cannot find the `ilAsqSolution` anywhere.

Reply:
@klees `ilAsqQuestionSolution`

Reply:
gets changed in the readme, sorry ;-)